react-frontend/src/profile/BookmarksCollection.js

```diff
@@ -1,38 +1,114 @@
-import React from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
 import { Link } from "react-router-dom";
+import axios from "axios";
+import emptyBookmark from "../assets/empty-bookmark.png";
+import filledBookmark from "../assets/filled-bookmark.png";
 
 function BookmarksCollection(props) {
-  console.log(props.bookmarks);
+  const { title, bookmarks } = props;
+  const [bookmarkStatus, setBookmarkStatus] = useState(
+    bookmarks.reduce((acc, bookmark) => {
+      acc[bookmark.listingID] = true; // All items are initially bookmarked
+      return acc;
+    }, {})
+  );
+
+  const toggleBookmark = async (listingID, listingTitle, userID) => {
+    const isBookmarked = bookmarkStatus[listingID];
+    if (isBookmarked) {
+      try {
+        await deleteBookmark(listingID, userID); // Pass the listingID to deleteBookmark function
+        setBookmarkStatus((prevStatus) => ({
+          ...prevStatus,
+          [listingID]: false,
+        }));
+      } catch (error) {
+        console.error("Error removing bookmark: ", error);
+      }
+    } else if (!isBookmarked) {
+      try {
+        setBookmarkStatus((prevStatus) => ({
+          ...prevStatus,
+          [listingID]: true,
+        }));
+        await createBookmark(listingID, listingTitle, userID);
+      } catch (error) {
+        console.error("Error creating bookmark: ", error);
+      }
+    }
+  };
+
+  const createBookmark = async (listingID, listingTitle, userID) => {
+    try {
+      await axios.post(
+        process.env.REACT_APP_BACKEND_LINK + `/listings/${listingID}/bookmark`, {
+          'userID': userID,
+          'listingID': listingID,
+          'title': listingTitle
+        }
+      );
+    } catch (error) {
+      console.error("Error creating bookmark: ", error);
+    }
+  };
+
+  const deleteBookmark = async (listingID, userID) => {
+    try {
+      await axios.delete(
+        process.env.REACT_APP_BACKEND_LINK + `/listings/${listingID}/bookmark`,
+        {
+          params: {
+            userID: userID,
+            listingID: listingID,
+          }
+        }
+      );
+    } catch (error) {
+      console.error("Error deleting bookmark: ", error);
+    }
+  };
 
   return (
     <div>
       <div className="vertical-center margin">
-        <h4>My {props.title}</h4>
+        <h4>My {title}</h4>
       </div>
       <div className="vertical-center margin">
         <div className="small-container bookmarks-height">
-          {props.bookmarks.length === 0 ? (
-              <p> There are no {props.title} </p>
-            ) : (
+          {bookmarks.length === 0 ? (
+            <p> There are no {title} </p>
+          ) : (
```

Comment:
> Seems let over from when this was a general collection component. If it's now just a bookmark collection component maybe change this to say bookmarks instead of using a prop.

True, I'll remove it since they are separate components now.